packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/dynamoDb-walkthrough.ts

```diff
@@ -61,490 +93,430 @@ async function updateWalkthrough(context, defaultValuesFilename, serviceMetadata
+        existingAttributeDefinitions.push(gsi.partitionKey);
+      }
+      if (gsi.sortKey) {
+        existingAttributeDefinitions.push(gsi.sortKey);
+      }
     });
-
-    if (attributeTypes[attributeAnswer[inputs[3].key]].indexable) {
-      indexableAttributeList.push(attributeAnswer[inputs[2].key]);
-    }
-
-    continueAttributeQuestion = await amplify.confirmPrompt('Would you like to add another column?');
   }
 
-  answers.AttributeDefinitions = attributeAnswers;
-
-  print.info('');
-  print.info(
-    'Before you create the database, you must specify how items in your table are uniquely organized. You do this by specifying a primary key. The primary key uniquely identifies each item in the table so that no two items can have the same key. This can be an individual column, or a combination that includes a primary key and a sort key.',
-  );
-  print.info('');
-  print.info('To learn more about primary keys, see:');
-  print.info(
-    'https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.CoreComponents.html#HowItWorks.CoreComponents.PrimaryKey',
-  );
-  print.info('');
-  // Ask for primary key
-
-  answers.KeySchema = [];
-  let partitionKeyName;
-  let partitionKeyType;
-
-  if (resourceName) {
-    ({ partitionKeyName } = defaultValues);
-    ({ partitionKeyType } = defaultValues);
-  } else {
-    const primaryKeyQuestion = {
-      type: inputs[4].type,
-      name: inputs[4].key,
-      message: inputs[4].question,
-      validate: amplify.inputValidation(inputs[3]),
-      choices: indexableAttributeList,
-    };
-
-    const partitionKeyAnswer = await inquirer.prompt([primaryKeyQuestion]);
-
-    partitionKeyName = partitionKeyAnswer[inputs[4].key];
+  if (!cliInputs.resourceName) {
+    throw new Error('resourceName not found in cli-inputs');
   }
 
-  answers.KeySchema.push({
-    AttributeName: partitionKeyName,
-    KeyType: 'HASH',
-  });
+  const { attributeAnswers, indexableAttributeList } = await askAttributeListQuestion(context, existingAttributeDefinitions);
 
-  // Get the type for primary index
+  cliInputs.gsi = await askGSIQuestion(context, indexableAttributeList, attributeAnswers, cliInputs.gsi);
+  cliInputs.triggerFunctions = await askTriggersQuestion(context, cliInputs.resourceName, cliInputs.triggerFunctions);
 
-  const primaryAttrTypeIndex = answers.AttributeDefinitions.findIndex(attr => attr.AttributeName === partitionKeyName);
+  cliInputsState.saveCliInputPayload(cliInputs);
 
-  partitionKeyType = answers.AttributeDefinitions[primaryAttrTypeIndex].AttributeType;
+  const stackGenerator = new DDBStackTransform(cliInputs.resourceName);
+  stackGenerator.transform();
 
-  usedAttributeDefinitions.add(partitionKeyName);
+  return cliInputs;
+}
 
-  let sortKeyName;
-  let sortKeyType;
+async function askTriggersQuestion(context: $TSContext, resourceName: string, existingTriggerFunctions?: string[]): Promise<string[]> {
+  const { amplify } = context;
```

Comment:
think we arent using amplify except for prompter

Reply:
Yup. Any helper for prompt which is recommended if not amplify.prompt?